tests/serverpod_test_server/test_integration/auth_core/auth_header_test.dart

```diff
@@ -73,5 +73,61 @@ void main() async {
       var receivedToken = await tokenInspectionCompleter.future;
       expect(receivedToken, key);
     });
+
+    test(
+        'then old clients passing keys in body should still work with new server code',
```

Comment:
Suggestion: Break apart the different test cases in this test.

We are validating multiple things in this test. 
Please break these scenarios apart so that it is clear what is failing.